common/FileUtils.java

```diff
@@ -415,5 +422,28 @@ public String downloadUrl(String url) throws IOException {
       throw new IllegalArgumentException(e);
     }
   }
+
+  /** Writes a proto to zip archive. */
+  public void writePrototxtToZip(Message proto, String absPathToZip, String prototxtFileName)
+      throws IOException {
+    String fileContent = TextFormat.printToUnicodeString(proto);
+    File tempPrototxt = File.createTempFile("temp_prototxt", ".prototxt");
+    BufferedWriter writer = new BufferedWriter(new FileWriter(tempPrototxt));
+    writer.write(fileContent);
+    writer.close();
+
+    Map<String, String> env = new HashMap<>();
+    env.put("create", "true");
+
+    URI uri = URI.create(String.format("jar:file:%s", joinPaths(absPathToZip)));
```

Comment:
Instead of writing the file to disk and then copying, and then deleting, how about we write it once, directly to the Zip filesystem?
See here:
https://github.com/google/startup-os/pull/491/commits/b0a19a72e2b30d39c3092cf6b33a2cef2b314d92#diff-929a018e48e8a9cf4ec4e3aa34ca311cR256
`proto.writeTo(Files.newOutputStream(fileSystem.getPath(expandHomeDirectory(path))));`
It would look something like:
`proto.writeTo(Files.newOutputStream(zipfs.getPath(prototxtFileName)));`